### PR TITLE
PSP-1481 Update frontend to React 17

### PR DESCRIPTION
**Pull request by asanchezr**

### CODE REVIEW

**Review by Fosol — APPROVED** [review]
Can you summary the Enzyme change?

File: Makefile [hunk 1]
```diff
@@ -68,8 +68,8 @@ rebuild: ## Build the local contains (n=service name) and then start them after
 clean: ## Removes all local containers, images, volumes, etc
 	@echo "$(P) Removing all containers, images, volumes for solution."
 	@docker-compose rm -f -v -s
-	@docker volume rm -f pims-frontend-node-cache
-	@docker volume rm -f pims-api-db-data
+	@docker volume rm -f psp-frontend-node-cache
+	@docker volume rm -f psp-api-db-data
```
Comment (asanchezr) [inline Makefile:72]: Updating local volumes here. They were updated in a previous PR

File: frontend/src/components/SearchBar/FilterBar.test.tsx [hunk 2]
```diff
@@ -1,12 +1,8 @@
-import Enzyme from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
```
Comment (asanchezr) [inline frontend/src/components/SearchBar/FilterBar.test.tsx:2]: These tests are not using enzyme (currently) so I removed the cruft

File: frontend/src/components/common/form/FastCurrencyInput.test.tsx [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
 import React from 'react';
 
 import { FastCurrencyInput } from './FastCurrencyInput';
 
-Enzyme.configure({ adapter: new Adapter() });
-
 describe('FastCurrencyInput', () => {
   it('fast currency input renders correctly', () => {
-    const { container } = render(
+    const { asFragment } = render(
```
Comment (asanchezr) [inline frontend/src/components/common/form/FastCurrencyInput.test.tsx:10]: This is the recommended way to do snapshot testing with `react-testing-library`

File: frontend/src/components/common/form/FastCurrencyInput.test.tsx [hunk 4]
```diff
@@ -66,7 +61,6 @@ describe('FastCurrencyInput', () => {
         )}
       </Formik>,
     );
-
-    expect(component.findByPlaceholderText('custom placeholder'));
+    expect(await findByPlaceholderText('custom placeholder')).toBeInTheDocument();
```
Comment (asanchezr) [inline frontend/src/components/common/form/FastCurrencyInput.test.tsx:64]: **findBy\*** returns a promise so we need to await here

File: frontend/src/components/common/form/ParentSelect.test.tsx [hunk 5]
```diff
@@ -58,15 +58,15 @@ it('changes to corresponding child value on click', async () => {
     </Formik>,
   );
   const test = container.querySelector('input[name="test"]');
-  await wait(() => {
+  await waitFor(() => {
```
Comment (asanchezr) [inline frontend/src/components/common/form/ParentSelect.test.tsx:61]: **wait** has been deprecated in the latest version of react-testing-library. Use **waitFor** instead

File: frontend/src/features/mapSideBar/containers/MotiInventoryContainer.test.tsx [hunk 6]
```diff
@@ -396,11 +396,12 @@ describe('MotiInventoryContainer functionality', () => {
...
       jest.clearAllMocks();
     });
+
     it('the cancel and submit buttons are disabled by default', async () => {
       const { findByText } = renderContainer({});
 
-      const cancelButton = await findByText('Cancel');
-      const submitButton = await findByText('Save');
+      const cancelButton = (await findByText('Cancel')).closest('button') as HTMLButtonElement;
+      const submitButton = (await findByText('Save')).closest('button') as HTMLButtonElement;
```
Comment (asanchezr) [inline frontend/src/features/mapSideBar/containers/MotiInventoryContainer.test.tsx:404]: This was finding a div within the button. Need to find the closest parent which is a **button**

File: frontend/src/setupTests.ts [hunk 7]
```diff
@@ -5,8 +5,12 @@
 import '@testing-library/jest-dom/extend-expect';
 import 'jest-styled-components';
 
+// Unofficial adapter for React 17 for Enzyme.
+// See https://github.com/enzymejs/enzyme/issues/2429 for details
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
 import { configure } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
```
Comment (asanchezr) [inline frontend/src/setupTests.ts:13]: I have moved all Enzyme configuration to this central file that gets executed prior to running the tests. 
No need to configure Enzyme on each test individually

File: frontend/src/utils/testUtils.ts [hunk 8]
```diff
@@ -30,45 +31,47 @@ export const fillInput = async (
     }
   }
 
-  await wait(() => {
```
Comment (asanchezr) [inline frontend/src/utils/testUtils.ts:33]: No need to use **wait** here anymore. The new version of react-testing-library wraps all calls to **fireEvent** with wait

**Comment by asanchezr** [comment]
We are now seeing that odd error in GitHub actions as well... Not sure how we will solve this
image

File: frontend/src/components/common/form/FastCurrencyInput.test.tsx (hunk 3 above)
Comment (jdtoombs) [reply frontend/src/components/common/form/FastCurrencyInput.test.tsx:10]: Ooh good to know

File: frontend/src/utils/testUtils.ts (hunk 8 above)
Comment (jdtoombs) [reply frontend/src/utils/testUtils.ts:33]: Liking these changes!

**Review by jdtoombs — APPROVED** [review]
Looks good, liking some of the testing changes with the new updates. Also still looking into this failing unit test.

**Pull request merged**